rllib/utils/checkpoints.py

```diff
@@ -29,15 +37,287 @@
 # 1.1: Same as 1.0, but has a new "format" field in the rllib_checkpoint.json file
 # indicating, whether the checkpoint is `cloudpickle` (default) or `msgpack`.
 
-# 1.2: Introduces the checkpoint for the new Learner API if the Learner api is enabled.
+# 1.2: Introduces the checkpoint for the new Learner API if the Learner API is enabled.
+
+# 2.0: Introduces the Checkpointable API for all components on the new API stack
+# (if the Learner-, RLModule, EnvRunner, and ConnectorV2 APIs are enabled).
 
 CHECKPOINT_VERSION = version.Version("1.1")
 CHECKPOINT_VERSION_LEARNER = version.Version("1.2")
+CHECKPOINT_VERSION_LEARNER_AND_ENV_RUNNER = version.Version("2.0")
+
+
+@PublicAPI(stability="alpha")
+class Checkpointable(abc.ABC):
```

Comment:
Dumb question at the beginning: Can't we directly derive from [`train.Checkpoint`](https://github.com/ray-project/ray/blob/bcbdcf2906408f3bac1f5a6af92e1e000bb94f59/python/ray/train/_checkpoint.py#L56) or at least its metaclass '_CheckpointMetaClass`?

Reply:
They are different concepts, I believe. train.Checkpoint is just describing the checkpoint itself, NOT the component that does get checkpointed.

I would probably interpret the new Checkpointable more as a competitor to tune.Trainable with its existing `save` and `restore` APIs and subsequent (save_checkpoint and load_checkpoint). We'll have to see, how to fit the Algorithm class itself into this new API and whether it's even possible. But for all the sub-components (which - in isolation - will never be used with Tune), this new API will unify and simplify a lot of things for our users. Especially when going to production/deployment.

Delivers an important promise of the new API stack: Plugability and re-usability of all components outside RLlib.